### ISSUE TITLE
Network messages error when switching to Combat from non-combat

### ISSUE THREAD

**Issue by TheRealNin** [Person8880/Shine#741]
When switching to a map that also runs Combat, Shine causes all clients to have the error ``Error: Different number of network messages on the Client from the Server`` and no one can join the server until the map switches.

If shine is removed from the mod list, this issue does not occur.

**Comment by Laaas** [comment]
I have experienced a similar issue, see [commit](https://github.com/woozasmoddingteam/Woozas-Potpourri/commit/ab87a36).

**Comment by Person8880** [comment]
This was caused by the change to plugin loading that moved network message registration to after all plugin files have loaded. Any plugins that declare `EnabledGamemodes`/`DisabledGamemodes` only on the server would end up registering network messages on the client but not the server. I generally only test with vanilla NS2, so I missed this, sorry!

I've fixed all the standard plugins to declare this restriction in `shared.lua` instead, and added a warning message to indicate if any other plugins are behaving in this way. If there are other plugins with the issue, either getting the author to move where they declare `EnabledGamemodes`/`DisabledGamemodes` to `shared.lua`, or using a `CanPluginLoad` hook to force disable the plugin in both realms will fix it.
